src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -241,6 +246,37 @@ impl StreamSink {
             }
         }
     }
+
+    fn infer_kv_log_store_table_catalog(&self) -> TableCatalog {
+        let mut table_catalog_builder =
+            TableCatalogBuilder::new(self.input.ctx().with_options().internal_table_subset());
+
+        // There are three pre-defined columns in the table: `epoch`, `seq_id` and `row_op`.
+        table_catalog_builder.add_column(&Field::with_name(EPOCH_COLUMN_TYPE, EPOCH_COLUMN_NAME));
+        table_catalog_builder.add_column(&Field::with_name(SEQ_ID_COLUMN_TYPE, SEQ_ID_COLUMN_NAME));
+        table_catalog_builder.add_column(&Field::with_name(ROW_OP_COLUMN_TYPE, ROW_OP_COLUMN_NAME));
+        let predefined_column_len = table_catalog_builder.columns().len();
+
+        // The table's pk is composed of `epoch` and `seq_id`.
+        table_catalog_builder.add_order_column(EPOCH_COLUMN_INDEX, OrderType::ascending());
+        table_catalog_builder
+            .add_order_column(SEQ_ID_COLUMN_INDEX, OrderType::ascending_nulls_last());
+        let read_prefix_len_hint = table_catalog_builder.get_current_pk_len();
+
+        let value_indices = table_catalog_builder.extend_columns(&self.sink_desc().columns);
+        table_catalog_builder.set_value_indices(value_indices);
+
+        // Modify distribution key indices based on the pre-defined columns.
+        let dist_key = self
+            .input
+            .distribution()
+            .dist_column_indices()
+            .iter()
+            .map(|idx| idx + predefined_column_len)
+            .collect_vec();
```

Comment:
Now we have an assumption that all the primary key should contains distribution key. we might need to use the `SEQ_ID_COLUMN` as the distribution key here with the `Serial` (refer https://github.com/risingwavelabs/risingwave/pull/8532 for more information)